packages/salesforcedx-lightning-lsp-common/src/providers/fileSystemDataProvider.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { FileStat, DirectoryEntry, WorkspaceConfig } from '../types/fileSystemTypes';
+
+/**
+ * Interface for file system operations
+ */
+export interface IFileSystemProvider {
+    getFileContent(uri: string): string | undefined;
+    getDirectoryListing(uri: string): DirectoryEntry[] | undefined;
+    getFileStat(uri: string): FileStat | undefined;
+    fileExists(uri: string): boolean;
+    directoryExists(uri: string): boolean;
+    updateFileContent(uri: string, content: string): void;
+    updateDirectoryListing(uri: string, entries: DirectoryEntry[]): void;
+    updateFileStat(uri: string, stat: FileStat): void;
+    updateWorkspaceConfig(config: WorkspaceConfig): void;
+}
+
+/**
+ * Manages file system data received from the LSP client
+ * This replaces direct file system access in the language server
+ */
+export class FileSystemDataProvider implements IFileSystemProvider {
```

Comment:
If you haven't already, I'd like you to get Pete's verdict on this.  It was my understanding the that the LS would need to run as its own process via a worker and what's passed in would be primitives.

Perhaps I misunderstood.  Or maybe these all run "inside" the extension rather than as a separate process and it's just "different" from apex that way?

Reply:
The custom endpoint that Kyle created to pass the std apex class zip from client to server gives us evidence shows that streaming is used on the client/server connection. Payload size may not be an issue for the connection, but transferring contents of workspace during init may impact perceived time to ready. 

The approach in the new LS is to use std LSP didOpen requests for each file found in the workspace. Madhur and I have discussed that option as well.

As you know, that approach may not be the correct one given the perf issues we have been discussing.